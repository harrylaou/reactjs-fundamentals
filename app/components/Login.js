import React, {Component} from 'react'
import {connect} from 'react-redux'
import {FormGroup, FormControl, Button} from 'react-bootstrap'
import withRouter from '../utils/WithRouter'

class Login extends Component {
  constructor () {
    super()
    this.state = {
      username: '',
      password: ''
    }
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleChange (name, event) {
    let change = {}
    change[name] = event.target.value
    this.setState(change)
  }

  handleSubmit () {
    if (this.state.username === 'demo' &&
        this.state.password === '1234') {

      // It's a temporary solution until you know more about redux
      localStorage.setItem('isAuthenticated', 1)
      this.props.router.push('/')
    }
  }

  render () {
    return (
      <form className="form-signin">
        <FormGroup>
          <h2 className="form-signin-heading">Please sign in</h2>
        </FormGroup>

        <FormGroup>
          <FormControl
            className="form-control"
            id="email"
            type="string"
            value={this.state.username}
            onChange={this.handleChange.bind(this, 'username')}
            placeholder="Enter email"
          />
          <FormControl
            className="form-control"
            id="password"
            type="password"
            value={this.state.password}
            onChange={this.handleChange.bind(this, 'password')}
            placeholder="Password"
          />
        </FormGroup>

        <Button
          bsSize="large"
          bsStyle="primary"
          onClick={this.handleSubmit}
        >
          Sign in
        </Button>
      </form>
    )
  }
}

export default withRouter(Login)
