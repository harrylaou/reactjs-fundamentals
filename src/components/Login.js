import React, { Component } from 'react'
import { connect } from 'react-redux'
import { FormGroup, FormControl, Button } from 'react-bootstrap'
import { withRouter } from 'react-router'

const Login = ({ username, password, handleSubmit, handleChange }) => (
  <form className="form-signin">
    <FormGroup>
      <h2 className="form-signin-heading">Please sign in</h2>
    </FormGroup>

    <FormGroup>
      <FormControl
        className="form-control"
        id="email"
        type="string"
        value={username}
        onChange={handleChange.bind(null, 'username')}
        placeholder="Enter email"
      />
      <FormControl
        className="form-control"
        id="password"
        type="password"
        value={password}
        onChange={handleChange.bind(null, 'password')}
        placeholder="Password"
      />
    </FormGroup>

    <Button
      bsSize="large"
      bsStyle="primary"
      onClick={handleSubmit}
    >
      Sign in
    </Button>
  </form>
)

Login.propTypes = {
  router: React.PropTypes.object,
  username: React.PropTypes.string,
  password: React.PropTypes.string,
  handleSubmit: React.PropTypes.func,
  handleChange: React.PropTypes.func,
}

export default Login
