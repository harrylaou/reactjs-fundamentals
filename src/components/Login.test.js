import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import Login from './Login'

import getMuiTheme from 'material-ui/styles/getMuiTheme'

describe('<Login />', () => {
  it('should render with default props', () => {
    const handleSubmit = jest.fn();
    const handleChange = jest.fn();
    const props = {
      username: 'demo',
      password: '12345',
      router: {},
      handleSubmit,
      handleChange
    }

    const wrapper = shallow(
      <Login {...props} />,
      {context: {
        muiTheme: getMuiTheme()
      }}
    )

    expect(shallowToJson(wrapper)).toMatchSnapshot()

    wrapper.find('Button').simulate('click')

    expect(handleSubmit).toBeCalled()

    wrapper.find('FormControl').at(0).simulate('change')

    expect(handleChange).toBeCalledWith('username')

    wrapper.find('FormControl').at(1).simulate('change')

    expect(handleChange).toBeCalledWith('password')
  })
})