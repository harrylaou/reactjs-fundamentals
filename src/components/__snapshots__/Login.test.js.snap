// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should render with default props 1`] = `
<form
  className="form-signin"
>
  <FormGroup
    bsClass="form-group"
  >
    <h2
      className="form-signin-heading"
    >
      Please sign in
    </h2>
  </FormGroup>
  <FormGroup
    bsClass="form-group"
  >
    <FormControl
      bsClass="form-control"
      className="form-control"
      componentClass="input"
      id="email"
      onChange={[Function]}
      placeholder="Enter email"
      type="string"
      value="demo"
    />
    <FormControl
      bsClass="form-control"
      className="form-control"
      componentClass="input"
      id="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value="12345"
    />
  </FormGroup>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="large"
    bsStyle="primary"
    disabled={false}
    onClick={[Function]}
  >
    Sign in
  </Button>
</form>
`;
